api/client.go

```diff
@@ -71,7 +74,12 @@ func (c *Client) Do(req *http.Request, v interface{}) (*http.Response, error) {
 	default:
 		if v != nil {
 			defer res.Body.Close()
-			if err := json.NewDecoder(res.Body).Decode(v); err != nil {
+
+			var bodyCopy bytes.Buffer
+			body := io.TeeReader(res.Body, &bodyCopy)
+
+			if err := json.NewDecoder(body).Decode(v); err != nil {
+				debug.Println("res.Body:", bodyCopy.String())
 				return nil, fmt.Errorf("error parsing API response - %s", err)
```

Comment:
OK, so it looks like this is when we do `api.Submissions()` after the fetch, the API is returning an error as JSON:

```
# http://exercism.io/api/v1/exercises?key=
{
  error: "Sorry—we can't figure out who you are. Double-check your API key in your exercism.io account page."
}
```

I'll change the API to return a zero-value for this call.